libservice/src/IpAddressNetlinkChecker.cpp

```diff
@@ -131,11 +142,96 @@ static std::optional<IPv4int> getMappedIPv4Addr(const in6_addr& addr) {
 	return ipv4Binary;
 }
 
+std::vector<IpAddressNetlinkChecker::Ipv6Interface> IpAddressNetlinkChecker::getIpv6Interfaces() const {
+	std::vector<Ipv6Interface> collectedIpv6Interfaces{};
+
+	ifaddrs *ifAddressStruct = nullptr;
+	if (getifaddrs(&ifAddressStruct) == 0) {
+		for (ifaddrs *ifa = ifAddressStruct; ifa != nullptr; ifa = ifa->ifa_next) {
+			if (ifa->ifa_addr != nullptr && ifa->ifa_addr->sa_family == AF_INET6) {
+				in6_addr interfaceIpv6Addr = reinterpret_cast<sockaddr_in6*>(ifa->ifa_addr)->sin6_addr;
+				in6_addr interfaceMask = reinterpret_cast<sockaddr_in6*>(ifa->ifa_netmask)->sin6_addr;
+
+				collectedIpv6Interfaces.emplace_back(Ipv6Interface{interfaceIpv6Addr, interfaceMask});
+			}
+		}
+		freeifaddrs(ifAddressStruct);
+	}
+
+	return collectedIpv6Interfaces;
+}
+
+IpAddressNetlinkChecker::ipv6Range IpAddressNetlinkChecker::parseIpv6Range(const std::string& range) {
+	ipv6Range rangeStruct;
+	size_t slashPos = range.find('/');
+	rangeStruct.ipv6Address = range.substr(0, slashPos);
+	rangeStruct.prefixLength = slashPos != std::string::npos ? std::stoi(range.substr(slashPos + 1)) : 0;
+	return rangeStruct;
+}
+
+bool IpAddressNetlinkChecker::isInRange(const in6_addr& addr, const std::string &range) {
+	IpAddressNetlinkChecker::ipv6Range rangeStruct = IpAddressNetlinkChecker::parseIpv6Range(range);
+
+	in6_addr rangeIpv6Address{};
+	inet_pton(AF_INET6, rangeStruct.ipv6Address.c_str(), &rangeIpv6Address);
+
+	// Create mask
+	in6_addr mask{};
+	for (size_t i = 0; i < sizeof(mask.s6_addr); i++) {
+		if (rangeStruct.prefixLength >= 8) {
+			mask.s6_addr[i] = 0xFF;
+			rangeStruct.prefixLength -= 8;
+		}
+		else if (rangeStruct.prefixLength > 0) {
+			mask.s6_addr[i] = (uint8_t)(0xFF << (8 - rangeStruct.prefixLength));
+			rangeStruct.prefixLength = 0;
+		}
+		else {
+			mask.s6_addr[i] = 0;
+		}
+	}
+
+	// Check mask and addr
+	in6_addr andResult{};
+	for (size_t i = 0; i < sizeof(andResult.s6_addr); ++i) {
+		andResult.s6_addr[i] = addr.s6_addr[i] & mask.s6_addr[i];
+	}
+
+	// Compare andResult with IPv6 address of the rangeStruct
+	return memcmp(&andResult, &rangeIpv6Address, sizeof(andResult)) == 0;
+}
+
+bool IpAddressNetlinkChecker::checkSubnet(const in6_addr& addrToCheck, const in6_addr& interfaceIpv6Addr, const in6_addr& interfaceMask) {
+	for (size_t i = 0; i < 4; ++i) {
```

Comment:
loop range is a magic number - is there a constant for that?

Reply:
I don't think so. I will calculate that value this way: `const auto s6Addr32ArraySize = sizeof(addrToCheck.s6_addr32) / sizeof(addrToCheck.s6_addr32[0]);`